### ISSUE TITLE
scan-manager-config.yaml should allow taints & node selectors

### ISSUE THREAD

**Issue by tunix** [trendmicro/cloudone-container-security-helm#10]
Hi,

As far as I understand, scan-manager pod dynamically spins up scanner jobs whenever a new deployment is detected and it does so using this config: https://github.com/trendmicro/cloudone-container-security-helm/blob/3068c4875dcfee12179bb7f35f610a5733a7b43d/templates/scan-manager-config.yaml

However, this config doesn't include any tolerations nor nodeSelector which prevents scanner jobs from starting when the cluster includes hybrid architecture nodes (i.e. Intel + ARM) or all nodes are tainted.

## When cluster has hybrid nodes

Pods either run on Intel or ARM nodes. Most of the time the image is exclusively set to a specific node group using either taints/tolerations or nodeSelectors (via `kubernetes.io/arch` for instance)

In such cases, scanner jobs fail when are scheduled to nodes with incorrect CPU architecture:

- deployment runs on ARM, scanner job is scheduled to AMD
- deployment runs on AMD, scanner job is scheduled to ARM

**Solution:** Scanner job should start with the same set of taint/tolerations or nodeSelectors configuration as the deployment

## When all nodes within the cluster are tainted

Scanner job config doesn't include any tolerations so the kube-scheduler cannot find a suitable node.

**Solution:** Scanner job should start with the same set of taint/tolerations or nodeSelectors configuration as the deployment

**Comment by zayaanmoez** [comment]
Hi @tunix,

Are you using multi-arch images? Would having the ability to specify tolerations or nodeSelectors on scanner pods through the helm values work? It would allow them to run when all nodes in the cluster are tainted but only work if the images are multi-arch

If you are not explicitly using multi-arch images, do you use taints/tolerations or nodeSelectors or both in the cluster with hybrid nodes?

**Comment by tunix** [comment]
Hi @zayaanmoez , we have 2 different clusters. On one, we have all nodes tainted. On the other, one but all the others are tainted.

We used to build all images as multi-arch images but we gave up on that as it makes our pipeline to execute longer unnecessarily.

I think you're right. Just having that configuration as an option would work in one of our clusters but not on the other. I think scanner jobs should definetely look at the source container and take such configuration whenever possible.

**Comment by zayaanmoez** [comment]
Hi @tunix, we have released 2.5.3 with support for inheriting node selectors and tolerations.
https://github.com/trendmicro/cloudone-container-security-helm/releases/tag/2.5.3

You can set the following in overrides to set tolerations and nodeSelectors on the scanner from the pod. It is recommended to define the filters to avoid inheriting any tolerations or node selectors for nodes with critical workloads.
```
nodeSelector:
  inheritNodeSelectorScanner: true # Inherit node selector from the owner resource (default: false)

  filterNodeSelectorScanner: # Only inherit node selector specified in the filter (default: all node selectors are inherited)
    kubernetes.io/arch: amd64
    kubernetes.io/arch: arm64

tolerations:
  inheritTolerationsScanner: true # Inherit tolerations from the owner resource (default: false)

  filterTolerationsScanner: # Only inherit tolerations specified in the filter (default: all tolerations are inherited)
  - key: kubernetes.io/arch 
    operator: Equal
    value: amd64
    effect: NoSchedule
  - key: kubernetes.io/arch 
    operator: Equal
    value: arm64
    effect: NoSchedule
```

More details: 
https://github.com/trendmicro/cloudone-container-security-helm?tab=readme-ov-file#configure-node-selectors-and-tolerations-for-the-container-security-components

**Comment by tunix** [comment]
We just tested this and it works! Thank you 🙏🏻
